testsupport/init.go

```diff
@@ -105,6 +107,76 @@ func waitForOperators(t *testing.T) {
 	t.Log("all operators are ready and in running state")
 }
 
+func getE2EServiceAccountToken(t *testing.T) string {
+	hostNs := os.Getenv(wait.HostNsVar)
+
+	apiConfigsa, err := clientcmd.NewDefaultClientConfigLoadingRules().Load()
+	require.NoError(t, err)
+
+	// creating another config which is used for creating resclient , client, for SA
+	//so that the main kubeconfig is not altered
+	restkubeconfig, err := util.BuildKubernetesRESTConfig(*apiConfigsa)
+	require.NoError(t, err)
+
+	sacl, err := client.New(restkubeconfig, client.Options{
+		Scheme: schemeWithAllAPIs(t),
+	})
+	require.NoError(t, err)
+
```

Comment:
these are already initialized in the `waitForOperators` function - you can pass them over as parameters

Reply:
i wanted to keep the restkubconfig seperate and hence initialised them seperately. so that the main kubeconfig is not affected, but i can try passing client, apiconfig